ams/client.go

```diff
@@ -176,12 +176,39 @@ func (c *Client) GetSubscriptions(organizationId string, statuses []string, size
 		}
 	}
 
+	if isSearchStrValid(searchParams.AccountUsername) {
+		queryBuilder = queryBuilder.And().Equals("creator.username", *searchParams.AccountUsername)
+	}
+
+	if isSearchStrValid(searchParams.Email) {
+		queryBuilder = queryBuilder.And().Equals("creator.email", *searchParams.Email)
+	}
+
+	if isSearchStrValid(searchParams.FirstName) {
+		queryBuilder = queryBuilder.And().Equals("creator.first_name", *searchParams.FirstName)
+	}
+
+	if isSearchStrValid(searchParams.LastName) {
+		queryBuilder = queryBuilder.And().Equals("creator.last_name", *searchParams.LastName)
+	}
+
 	query := queryBuilder.Build()
 
+	orderBy, err := buildOrderBy(searchParams.Sort, searchParams.SortOrder)
+	if err != nil {
+		return nil, &ClientError{
+			Message: err.Error(),
+			StatusCode: http.StatusBadRequest,
+			OrgId: organizationId,
+			AmsOrgId: amsOrgId,
+		}
+	}
+
 	start := time.Now()
 	req := c.client.AccountsMgmt().V1().Subscriptions().List().
 		Search(query).
 		Parameter("fetchAccounts", true).
+		Parameter("orderBy", orderBy).
```

Comment:
so right now this actually breaks because "orderBy" does not support searching on any of the `creator.*` fields. So to merge this PR we have to disable the sorting params for now

Reply:
removed support for sort, we can do that in another ticket

Reply:
https://issues.redhat.com/browse/RHCLOUD-28509